src/ledger/LedgerTxn.cpp

```diff
@@ -2953,11 +2972,37 @@ LedgerTxnRoot::dropTTL(bool rebuild)
 uint32_t
 LedgerTxnRoot::prefetch(UnorderedSet<LedgerKey> const& keys)
 {
-    return mImpl->prefetch(keys);
+    UnorderedMap<LedgerKey, UnorderedSet<Hash>> lkToTx;
+    UnorderedMap<Hash, uint32_t> txReadBytes;
+    UnorderedSet<LedgerKey> notLoaded;
+    return prefetchWithLimits(keys, lkToTx, txReadBytes, notLoaded);
+}
+uint32_t
+LedgerTxnRoot::prefetchWithLimits(
+    UnorderedSet<LedgerKey> const& keys,
+    UnorderedMap<LedgerKey, UnorderedSet<Hash>>& lkToTx,
+    UnorderedMap<Hash, uint32_t>& txReadBytes,
+    UnorderedSet<LedgerKey>& notLoaded)
+
+{
+    return mImpl->prefetchWithLimits(keys, lkToTx, txReadBytes, notLoaded);
 }
 
 uint32_t
 LedgerTxnRoot::Impl::prefetch(UnorderedSet<LedgerKey> const& keys)
+{
+    UnorderedMap<LedgerKey, UnorderedSet<Hash>> lkToTx;
+    UnorderedMap<Hash, uint32_t> txReadBytes;
+    UnorderedSet<LedgerKey> notLoaded;
+    return prefetchWithLimits(keys, lkToTx, txReadBytes, notLoaded);
+}
+
+uint32_t
+LedgerTxnRoot::Impl::prefetchWithLimits(
```

Comment:
I don't fully understand the prefetch flows; is there any way to avoid logic duplication between this and `Bucket.cpp`?